core/src/main/scala/io/delta/tables/DeltaOptimizeBuilder.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.tables
+
+import org.apache.spark.sql.{DataFrame, Dataset, SparkSession}
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.delta.commands.OptimizeTableCommand
+import org.apache.spark.sql.delta.util.AnalysisHelper
+
+
+/**
+ * Builder class for constructing OPTIMIZE command and executing.
+ *
+ * @param sparkSession SparkSession to use for execution
+ * @param tableIdentifier Id of the table on which to
+ *        execute the optimize
+ * @since 1.3.0
+ */
+class DeltaOptimizeBuilder(
+    sparkSession: SparkSession,
+    tableIdentifier: String) extends AnalysisHelper {
+  private var partitionFilter: Option[String] = None
+
+  /**
+   * Apply partition filter on this optimize command builder to limit
+   * the operation on selected partitions.
+   * @param partitionFilter The partition filter to apply
+   * @return [[DeltaOptimizeBuilder]] with partition filter applied
+   */
+  def partitionFilter(partitionFilter: String): DeltaOptimizeBuilder = {
```

Comment:
Hi @Kimahriman, had an offline conversation with @tdas. He mentioned one very good point about keeping the names same as SQL. In SQL we have `where` to select partitions. Renaming this method to `def where(partitionFilter: String)` to keep it sync with the SQL.

We follow this pattern in other APIs as well. For example in Merge: SQL has `WHEN MATCHED` and in Scala/Python we have similar named method `whenMatched`.

Let me know if there are any concerns with the rename. I can make locally change and put it into the merge queue.

Reply:
I have no problem with that, just let me know if you want me to change anything or if you'll handle it

Reply:
Thanks @Kimahriman, its ok I can make the change.